classes/cron_manager.php

```diff
@@ -63,39 +62,29 @@ function() use ($profiler, $timer) {
      *
      * We then check for a new task with the current sample.
      *
-     * @param \ExcimerProfiler $profiler
+     * @param object $profiler
      */
-    public static function on_interval(\ExcimerProfiler $profiler): void {
-        $log = $profiler->flush();
+    public static function on_interval(object $profiler): void {
+        $log = $profiler->profiler->flush();
 
-        $log->rewind();
-        if (!$log->valid()) {
-            debugging('on_interval called with no profile');
-            return;
-        }
-        $sample = $log->current();
-        $log->next();
-        if ($log->valid()) {
-            debugging('extra profile sample found in on_interval');
-        }
-
-        $taskname = self::findtaskname($sample);
-        $currenttime = microtime(true);
+        foreach ($log as $sample) {
+            $taskname = self::findtaskname($sample);
+            $currenttime = $profiler->started + $sample->getTimestamp();
+            if (self::$currenttask && (self::$currenttask->name != $taskname)) {
+                self::$currenttask->process($currenttime);
+                self::$currenttask = null;
+            }
 
-        if (self::$currenttask && (self::$currenttask->name != $taskname)) {
-            self::$currenttask->process($currenttime);
-            self::$currenttask = null;
-        }
+            if ($taskname && (self::$currenttask == null)) {
+                self::$currenttask = new task_samples($taskname, self::$sampletime);
+            }
 
-        if ($taskname && (self::$currenttask == null)) {
-            self::$currenttask = new task_samples($taskname, self::$sampletime);
-        }
+            if (self::$currenttask) {
+                self::$currenttask->add_sample($sample);
+            }
 
-        if (self::$currenttask) {
-            self::$currenttask->add_sample($sample);
+            self::$sampletime = $currenttime;
```

Comment:
It looks like this isnt necessary inside a loop, instead it could be set outside the loop, based on the context of the most recent log sample?